projects/igniteui-angular/src/lib/query-builder/query-builder.directives.ts

```diff
@@ -21,3 +23,44 @@ import { Directive, TemplateRef } from '@angular/core';
 export class IgxQueryBuilderSearchValueTemplateDirective {
     constructor(public template: TemplateRef<any>) { }
 }
+
+/** @hidden @internal */
+@Directive({
+    selector: '[igxFieldValidators]',
+    standalone: true
+})
+export class IgxFieldValidatorDirective implements OnChanges {
+    private picker: any;
+    constructor(private element: ElementRef,
+        @Host() @Self() @Optional() private datePicker : IgxDatePickerComponent,
+        @Host() @Self() @Optional() private timePicker : IgxTimePickerComponent,
+        @Host() @Self() @Optional() private dateTimePicker : IgxDateTimeEditorDirective
+    ) {
+        this.picker = this.datePicker || this.timePicker || this.dateTimePicker;
+    }
+
+    @Input()
+    public validators: IFieldValidator[] = [];
+
+    public ngOnChanges() {
+        if (this.validators) {
+            if(this.picker) {
+                this.validators.forEach(validator => {
+                    if (validator.type === 'mindate') {
+                        this.picker.minValue = validator.value;
+                    } else if (validator.type === 'maxdate') {
+                        this.picker.maxValue = validator.value;
+                    }
+                });
+            } else {
+                this.validators.forEach(validator => {
+                    if (validator.type === 'required') {
+                        this.element.nativeElement.required = true;
+                    } else if (!validator.type.includes('date')) {
+                        this.element.nativeElement.setAttribute(validator.type, validator.value);
+                    }
+                });
+            }
+        }
```

Comment:
With this if-else structure the pickers (including dateTimeEditor) would never get the required attribute. 
I would suggest using a switch for better readability:
```suggestion
        if (this.validators) {
            this.validators.forEach(validator => {
                switch (validator.type) {
                    case 'required':
                        this.element.nativeElement.required = true;
                        break;
                    case 'mindate':
                        if (this.picker)
                            this.picker.minValue = validator.value;
                        break;
                    case 'maxdate':
                        if (this.picker)
                            this.picker.maxValue = validator.value;
                        break;
                    default:
                        this.element.nativeElement.setAttribute(validator.type, validator.value);
                        break;      
                }
            });
        }
```